torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -461,6 +521,7 @@ void ProcessGroupNCCL::WorkNCCL::synchronizeInternal(
       AT_CUDA_CHECK(cudaDeviceSynchronize());
     }
   }
+  markRankNotStuck(rankStatusHelperUnowned_, rank_ % at::cuda::getNumGPUs());
```

Comment:
cache `at::cuda::getNumGPUs()`? Or maybe its stored as a variable in this file already, not sure.